From 7fda4345bdabafa896f4a632990c20d91076e98d Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 27 Aug 2023 19:48:28 +0200
Subject: [PATCH] load_library: handle AttributeError as well

Looks like with Python 3.12 getattr on cdll now correctly raises
AttributeError instead of OSError, so catch that too.
This fixes the tests in case libxine is missing.
---
 quodlibet/util/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quodlibet/util/__init__.py b/quodlibet/util/__init__.py
index 40f168e492..4db47ad8c4 100644
--- a/quodlibet/util/__init__.py
+++ b/quodlibet/util/__init__.py
@@ -961,7 +961,7 @@ def load_library(names, shared=True):
 
         try:
             return load_func(dlopen_name), name
-        except OSError as e:
+        except (OSError, AttributeError) as e:
             errors.append(str(e))
 
     raise OSError("\n".join(errors))
